src/v2/Apps/Partner/Components/PartnerViewingRooms/ViewingRooms.tsx

```diff
@@ -0,0 +1,43 @@
+import { Column, GridColumns, Text } from "@artsy/palette"
+import React from "react"
+import { createFragmentContainer, graphql } from "react-relay"
+import { ViewingRoomCardFragmentContainer } from "./ViewingRoomCard"
+import { ViewingRooms_edges } from "v2/__generated__/ViewingRooms_edges.graphql"
+
+interface ViewingRoomsProps {
+  edges: ViewingRooms_edges
+  eventTitle: string
+}
+
+const ViewingRooms: React.FC<ViewingRoomsProps> = ({ edges, eventTitle }) => (
+  <>
+    <Text color="black" variant="title" mb={6}>
+      {eventTitle}
+    </Text>
+
+    <GridColumns mb={6} gridRowGap={[2, 4]}>
+      {edges.map(({ node: viewingRoom }) => {
+        if (!viewingRoom) return
```

Comment:
discussion: Something about this is off to me, I'm not sure if it's the two return statements back-to-back or that we're returning `undefined` in the early exit and hoping that `<GridColumns>` will handle that correctly.

I would feel better if we filtered out the falsey `node`s from `edges` earlier, maybe before returning this component?

```js
const ViewingRooms: React.FC<ViewingRoomsProps> = ({ edges, eventTitle }) => {
  const viewingRooms = edges.filter(node => node.viewingRoom)
  return (
    <>
      <Text color="black" variant="title" mb={6}>
        {eventTitle}
      </Text>

      <GridColumns mb={6} gridRowGap={[2, 4]}>
        {viewingRooms.map((viewingRoom) => {
  		  return (
            <Column key={viewingRoom.internalID} span={[6, 6, 3, 3]}>
              <ViewingRoomCardFragmentContainer viewingRoom={viewingRoom} />
            </Column>
          )
        })}
  ...
}
```
What do you think? It iterates through the `edges` array twice instead of once, but we don't have to worry about falsey `viewingRoom` values affecting `<GridColumns>`

Reply:
I've done this before) in: https://github.com/artsy/force/blob/ef6ee3e70f572cdb24d67992b8770078b21f87f6/src/v2/Apps/Partner/Routes/ViewingRooms/index.tsx#L24

This is fine?

Reply:
Yeah, that works for me!